### PR TITLE
feat: add generation of the moved parameter from previous input files

### PR DESCRIPTION
**Pull request by Demian222**

### CODE REVIEW

File: src/main/js/generateStatic.js [hunk 1]
```diff
@@ -3,40 +3,47 @@ import fsExtra from 'fs-extra'
...
+ * @param contexts
  * @param _output
+ * @param basePrefix
+ * @param intermediate
  */
-export const genStatics = async (
-  docs,
-  dirs,
+export const generateStatics = async (
+  contexts,
```
Comment (antongolub) [inline src/main/js/generateStatic.js:18]: `moved` field resolver should be pulled up to the upper layer.

**Review by antongolub — CHANGES_REQUESTED** [review]

File: src/main/js/generateMdAssets.js [hunk 2]
```diff
@@ -51,42 +51,51 @@ export const genMdAssets = (doc, temp) => {
   })
 }
 
-export const genParamMove = (file, doc, intermediate, date) => {
+export const genParamMove = (contexts, intermediate, autoscope) => {
+  if (!autoscope) return contexts
+  return contexts.map((context) => {
+    const { file, data, date } = context
+
+    const validDate = new Date(date)
```
Comment (antongolub) [inline src/main/js/generateMdAssets.js:59]:
```ts
_date = new Date()

const prevCtx = intermediate.find(...)

```

File: src/main/js/generateMdAssets.js [hunk 3]
```diff
@@ -51,42 +51,51 @@ export const genMdAssets = (doc, temp) => {
...
+    if (isNaN(validDate.getDate())) {
+      throw new Error(`${file} - invalid date`)
+    }
+    const findValue = intermediate.find((i) => {
+      return i.dir === path.dirname(file)
+    })
+    if (findValue && findValue.date < validDate) {
+      return { ...context, data: modifyData(data, findValue) }
+    }
+    const interData = {}
```
Comment (antongolub) [inline src/main/js/generateMdAssets.js:69]: reduce

File: src/main/js/generateMdAssets.js [hunk 4]
```diff
@@ -51,42 +51,51 @@ export const genMdAssets = (doc, temp) => {
...
+    data.data.forEach((item) => {
+      interData[item.name.toLowerCase()] = item.ring.toLowerCase()
+    })
+    if (!findValue) {
+      intermediate.push({
+        dir: path.dirname(file),
+        data: interData,
+        date: validDate,
```
Comment (antongolub) [inline src/main/js/generateMdAssets.js:77]: this value is not a string. This may affect the static renderer.

**Review by antongolub — CHANGES_REQUESTED** [review]

File: src/main/js/generateMdAssets.js (hunk 4 above)
Comment (Demian222) [reply src/main/js/generateMdAssets.js]: date string is in data.meta.date

**Review by antongolub — APPROVED** [review]
LGTM

**Pull request merged**